src/CMake/FindZlib.cmake

```diff
@@ -26,40 +26,87 @@
 #   Kathleen Biagas, Tue May 28 09:08:56 PDT 2019
 #   Since we require zlib from build_visit, no longer need HAVE_ZLIB_H
 #
+#   Kathleen Biagas, Fri May 31, 2024
+#   Modified to create an INTERFACE target via blt_import_library.
+#   This allows zlib to be a native CMake target, and will also be accounted
+#   for in the generated export set.
+#
 #****************************************************************************/
 
 # Use the ZLIB_DIR hint from the config-site .cmake file
 
-if (WIN32)
-    if(ZLIB_LIB_NAME)
-        SET_UP_THIRD_PARTY(ZLIB LIBS ${ZLIB_LIB_NAME})
+if(ZLIB_DIR)
+
+    find_path(_zlib_INCLUDE_DIR zlib.h
+              PATHS ${ZLIB_DIR}
+              PATH_SUFFIXES include
+              NO_DEFAULT_PATH)
+
+    find_library(_zlib_LIBRARY
+             NAMES zlib zlib1 z
+             PATHS ${ZLIB_DIR}
+             PATH_SUFFIXES lib lib64
+             NO_DEFAULT_PATH)
+
+    find_package_handle_standard_args(ZLIB DEFAULT_MSG
+        _zlib_INCLUDE_DIR
+        _zlib_LIBRARY)
+
+    if(ZLIB_FOUND)
+        get_filename_component(libz ${_zlib_LIBRARY} NAME)
+        set(_zlib_DLL)
+
+        ####
+        # until everything is updated to use the new target
+        get_filename_component(ZLIB_LIBRARY_DIR ${_zlib_LIBRARY} PATH)
+        set(ZLIB_LIBRARY ${_zlib_LIBRARY})
+        set(ZLIB_LIB ${libz})
+        set(ZLIB_INCLUDE_DIR ${_zlib_INCLUDE_DIR})
+        ####
+
+
+        blt_import_library(
+            NAME        zlib
+            INCLUDES    $<BUILD_INTERFACE:${_zlib_INCLUDE_DIR}>
+                        $<INSTALL_INTERFACE:${VISIT_INSTALLED_VERSION_INCLUDE}/zlib/include>
+            LIBRARIES   $<BUILD_INTERFACE:${_zlib_LIBRARY}>
+            EXPORTABLE  ON)
+
+
+        # CMake doesn't prepend ${_IMPORT_PREFIX} in the generated export
+        # set for INTERFACE libs, so "${IMPORT_PREFIX}" needs to be
+        # explicitly added the the INSTALL_INTERFACE, and escaped so it
+        # doesn't get evaluated.
+        #
+        # Also, it seems if the INSTALL_INTERFACE is used in the
+        # blt_import_library,  the ${_IMPORT_PREFIX} is stripped, even
+        # if it is escaped, so it appears to be getting evaluated.
+        # That's why it is added separately here instead of being added
+        # to the LIBRARIES above.
+        target_link_libraries(zlib INTERFACE
```

Comment:
Definitely a CMake problem. They do not add `_INSTALL_PREFIX` to the beginning of the link libraries listing when generating the export set (visitTargets.cmake).  This doesn't affect the build, but will cause problems when we start using the export sets (eg building a plugin vs installed VisIt). Not sure why this behavior is different than how INCLUDE DIRECTORIES are handled.  

Perhaps exacerbated by the way BLT processes the DEPENDS_ON list.

What I know for certain, we have to manually prepend `${_INSTALL_PREFIX}` and add the '\\' escape before the `$` so that CMake doesn't attempt to exapand the variable.